crawl-ref/source/mon-place.cc

```diff
@@ -964,82 +742,19 @@ monster* place_monster(mgen_data mg, bool force_pos, bool dont_place)
             if (map_masked(mg.pos, mg.map_mask))
                 continue;
 
-            // Let's recheck these even for PROX_NEAR_STAIRS, just in case.
-            // Check proximity to player.
-            proxOK = true;
-
-            switch (mg.proximity)
-            {
-            case PROX_ANYWHERE:
-                if (grid_distance(you.pos(), mg.pos) < 2 + random2(3))
-                    proxOK = false;
-                break;
-
-            case PROX_CLOSE_TO_PLAYER:
-            case PROX_AWAY_FROM_PLAYER:
-                // If this is supposed to measure los vs not los,
-                // then see_cell(mg.pos) should be used instead. (jpeg)
-                close_to_player = (grid_distance(you.pos(), mg.pos) <=
-                                   LOS_RADIUS);
-
-                if (mg.proximity == PROX_CLOSE_TO_PLAYER && !close_to_player
-                    || mg.proximity == PROX_AWAY_FROM_PLAYER && close_to_player)
-                {
-                    proxOK = false;
-                }
-                break;
-
-            case PROX_NEAR_STAIRS:
-                if (pval == 2) // player on stairs
-                {
-                    if (mons_class_base_speed(mg.cls) == 0)
-                    {
-                        proxOK = false;
-                        break;
-                    }
-                    // Swap the monster and the player spots, unless the
-                    // monster was generated in lava or deep water.
-                    if (grd(mg.pos) == DNGN_LAVA
-                        || grd(mg.pos) == DNGN_DEEP_WATER)
-                    {
-                        proxOK = false;
-                        break;
-                    }
-
-                    // You can't be shoved if you're caught in a net.
-                    if (you.caught())
-                    {
-                        proxOK = false;
-                        break;
-                    }
-
-                    shoved = true;
-                    coord_def mpos = mg.pos;
-                    mg.pos         = you.pos();
-                    you.moveto(mpos);
-                }
-                proxOK = (pval > 0);
-                break;
-            }
-
-            if (!proxOK)
-                continue;
-
-            // Cool.. passes all tests.
             break;
-        } // end while... place first monster
+        }
     }
+    // Sanity check that the specified position is valid.
     else if (!_valid_monster_generation_location(mg) && !dont_place)
-    {
-        // Sanity check that the specified position is valid.
-        return 0;
-    }
+        return nullptr;
 
-    monster* mon = _place_monster_aux(mg, 0, place, force_pos, dont_place);
+    monster* mon
+        = _place_monster_aux(mg, nullptr, place, force_pos, dont_place);
 
     // Bail out now if we failed.
```

Comment:
Comment is extraneous IMO. It's already quite clear what the next two lines do and why.

Reply:
Will remove.